enterprise/cmd/frontend/internal/authz/resolvers/resolver.go

```diff
@@ -365,10 +366,13 @@ func (r *Resolver) CancelPermissionsSyncJob(ctx context.Context, args *graphqlba
 	err = r.db.PermissionSyncJobs().CancelQueuedJob(ctx, reason, syncJobID)
 	// We shouldn't return an error when the job is already processing or not found
 	// by ID (might already be cleaned up).
-	if err != nil && !errcode.IsNotFound(err) {
+	if err != nil {
+		if errcode.IsNotFound(err) {
+			return pointer.String("Permissions sync job is already dequeued and cannot be canceled."), nil
```

Comment:
```suggestion
			return pointer.String("No job that can be canceled found."), nil
```

Everything else is an assumption :)

Reply:
yeah, good point. I think in this case I'd personally change the error message on the client and display a custom message. This message here, that's the API message to me. If someone uses the API to cancel a job with an invalid ID we can't tell them "job is already dequeued" ,right?

Reply:
that's a nice tradeoff! 👍🏻